src/MLJCatBoostInterface.jl

```diff
@@ -28,6 +28,40 @@ function mlj_to_kwargs(model)
                             for name in fieldnames(typeof(model)))
 end
 
+"""
+    get_dtype_feature_ix(X, dtype)
+
+Get the index of the columns with a specific dtype
+
+Parameters
+----------
+- `X`: Table
+- `dtype`: DataType 
+
+Returns
+-------
+- `Vector{Int64}`
+"""
+function get_dtype_feature_ix(X, dtype)
+    return findall(eltype.(MMI.scitype.(eachcol(X))) .<: dtype)
+end
+
+"""
+Get cat features for model
+get_cat_features
+"""
+function prepare_input(X)
+    order_factor_ix = get_dtype_feature_ix(X, OrderedFactor)
+    for col in names(X)[order_factor_ix]
+        X[:, col] = MMI.int(X[:, col])
+    end
+
```

Comment:
As I understand the later call to this function, we should only be assuming that `X` satisfies `Tables.istable(X) =true`, ie, we don't know what kind of table this is. So I'd think `names(X)` should probably be something like `MMI.schema(X).names` or `Tables.columnnames(columns(X))` or something like that. And you will not be able to mutate columns using the line at 56, but (unfortunately) must create a new object (eg a named-tuple of vectors, which is also a Tables.jl table) by grabbing each column using the [Tables API](https://tables.juliadata.org/stable/#Using-the-Interface-(i.e.-consuming-Tables.jl-compatible-sources)).

Reply:
resolved